### PR TITLE
DS-2007: add batchSize commandline option to bin/dspace cleanup;  make it continue past exceptions

### PR DESCRIPTION
**Pull request by akinom**
add option to control commit batch size to    bin/dspace cleanup 

cleaned BitstreamStorageManager code up a bit while moving exception handling inside the loop that goes over deleted bitstreams 

https://jira.duraspace.org/browse/DS-2007

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/storage/bitstore/BitstreamStorageManager.java [hunk 1]
```diff
@@ -561,150 +563,155 @@ public static void delete(Context context, int id) throws SQLException
...
         // Aborting will leave the DB objects around, even if the
-        // bitstreams are deleted. This is OK; deleting them next
-        // time around will be a no-op.
-        catch (SQLException sqle)
-        {
+        // bitstreams are deleted. This is OK; they will be deleted
+        // next time around
+        catch (SQLException sqle) {
+            System.err.println("Error: " + sqle.getMessage());
             if (verbose)
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/storage/bitstore/BitstreamStorageManager.java]: While you are at it, why not remove the condition test here and always report the exception regardless of 'verbose'?  That would take care of DS-3074 as well.  It is strange that we don't report an event which aborts the command.

File: dspace-api/src/main/java/org/dspace/storage/bitstore/BitstreamStorageManager.java [hunk 2]
```diff
@@ -561,150 +563,155 @@ public static void delete(Context context, int id) throws SQLException
...
-            {
-                System.err.println("Error: " + sqle.getMessage());
-            }
-            context.abort();
-            throw sqle;
-        }
-        catch (IOException ioe)
-        {
+                System.err.println(sqle);
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/storage/bitstore/BitstreamStorageManager.java]: Why do we print the exception twice if 'verbose'?

File: dspace-api/src/main/java/org/dspace/storage/bitstore/BitstreamStorageManager.java [hunk 3]
```diff
@@ -561,150 +563,155 @@ public static void delete(Context context, int id) throws SQLException
...
+            success = false;
+        } catch (IOException ioe) {
+            System.err.println("Error: " + ioe.getMessage());
             if (verbose)
-            {
-                System.err.println("Error: " + ioe.getMessage());
-            }
-            context.abort();
-            throw ioe;
+                System.err.println(ioe);
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/storage/bitstore/BitstreamStorageManager.java]: Why do we print the exception twice if 'verbose'?

**Comment by mwoodiupui** [comment]
Can this be updated in time for 6.0?

**Comment by mwoodiupui** [comment]
The documentation is in readme.wiki, apparently to be copied into Confluence when this goes in.

**Comment by akinom** [comment]
I fixed the code so it merges successfully
and reset the head of  akinom:prq_cleanup     with   git reset —hard to the merged code

comparing the branches directly  gives a short list of changes
https://github.com/DSpace/DSpace/compare/master...akinom:prq_cleanup?expand=1

But the linked ‘View pull request<https://github.com/DSpace/DSpace/pull/539>’  shows a crazy long list of changes

I have no idea why that happens - so I don’t know how to fix this

Monika


—
Monika Mevenkamp
Digital Repository Infrastructure Developer
Princeton University
Phone: 609-258-4161
Skype: mo-meven

On Mar 16, 2016, at 3:20 PM, Mark H. Wood <notifications@github.com<mailto:notifications@github.com>> wrote:


Can this be updated in time for 6.0?

**Comment by tdonohue** [comment]
Closing as this has been replaced by #1337

**Pull request closed without merging**